main.go

```diff
@@ -6,55 +6,142 @@ import (
 	"os/signal"
 	"time"
 
+	"github.com/ONSdigital/dp-api-clients-go/dataset"
+	"github.com/ONSdigital/dp-api-clients-go/filter"
+	"github.com/ONSdigital/dp-api-clients-go/hierarchy"
+	"github.com/ONSdigital/dp-api-clients-go/renderer"
+	"github.com/ONSdigital/dp-api-clients-go/search"
 	"github.com/ONSdigital/dp-frontend-filter-dataset-controller/config"
 	"github.com/ONSdigital/dp-frontend-filter-dataset-controller/routes"
+	health "github.com/ONSdigital/dp-healthcheck/healthcheck"
 	"github.com/ONSdigital/go-ns/handlers/collectionID"
-	"github.com/ONSdigital/go-ns/log"
 	"github.com/ONSdigital/go-ns/server"
+	"github.com/ONSdigital/log.go/log"
 	"github.com/gorilla/mux"
 )
 
+// App version informaton retrieved on runtime
+var (
+	// BuildTime represents the time in which the service was built
+	BuildTime string
+	// GitCommit represents the commit (SHA-1) hash of the service that is running
+	GitCommit string
+	// Version represents the version of the service that is running
+	Version string
+)
+
 func main() {
 	log.Namespace = "dp-frontend-filter-dataset-controller"
+
+	signals := make(chan os.Signal, 1)
+	signal.Notify(signals, os.Interrupt, os.Kill)
+
+	ctx := context.Background()
+
 	cfg, err := config.Get()
 	if err != nil {
-		log.Error(err, nil)
+		log.Event(ctx, "unable to retrieve service configuration", log.Error(err))
 		os.Exit(1)
 	}
-	ctx := context.Background()
 
-	log.InfoCtx(ctx, "got service configuration", log.Data{"config": cfg})
+	timeout, err := time.ParseDuration(cfg.CriticalTimeout)
```

Comment:
You could also include this conversion as part of the `Get()` function. If when called the time values cannot be parsed to duration then return an error. It would have having to individually check each field.

Reply:
In the same line: envconfig supports automatically parsing time.Duration values:
https://github.com/kelseyhightower/envconfig
I would suggest to define the type as time.Duration inside the config struct, instead of doing time.ParseDuration